pkg/source/pulsar.go

```diff
@@ -139,7 +139,7 @@ func (p *PulsarReaderSource) Capture(cp cursor.Checkpoint) (changes chan Change,
 			}
 		}
 
-		change = Change{Checkpoint: checkpoint, Message: m}
+		change = Change{Checkpoint: checkpoint, Message: m, HasNext: p.reader.HasNext()}
```

Comment:
I am not sure that using `p.reader.HasNext()` is a good idea. This method is kind of unstable.
Could we try adding a `default` block to the `select` statement here: https://github.com/rueian/pgcapture/blob/459cefef9f221c49413c3b9cc3b6ee56c7c8ada7/pkg/sink/main.go#L51-L61 to detect if there are no more messages?

Reply:
I feel that if we don't determine hasNext at the source side, then having to modify ApplyFn or using a mark change might seem a bit hacky. 
After discussing with @benjamin99 , maybe we can first try using the current implementation.

Reply:
After discussing with @rueian , we decide to add `sourceChangesSize` in `ApplyFn` to let sink to decide that how to handle this situation. cc @benjamin99